wire-runtime/src/commonMain/kotlin/com/squareup/wire/ProtoReader.kt

```diff
@@ -131,6 +131,28 @@ class ProtoReader(private val source: BufferedSource) {
     endMessageAndGetUnknownFields(token)
   }
 
+  /** Reads and returns the length of the next message in a length-delimited stream. */
+  @Throws(IOException::class)
+  fun nextLengthDelimited(): Int {
+    check(state == STATE_TAG || state == STATE_LENGTH_DELIMITED) {
+      "Unexpected call to nextDelimited()"
+    }
```

Comment:
It's not clear this is useful. Could eliminate and only have the one function.

Reply:
Could we make it clearer by telling which state we're in? I don't mind whether we keep it or not

Reply:
The message is the same as other places in this file. I can update them all separately to include the current state.